### DISCUSSION TITLE
Is audio/video download  supported in pyodide?

### DISCUSSION THREAD

**Discussion by bigMarxOne** [pyodide/pyodide#4842] [General]
I am a beginner in pyodide. Now I am developing a browser plugin where audio files of videos are needed to be download for AI processing in pyodide. 

When I tryed to use  a python open source tool youtube--dl for it, some issues appeared as following. The same issues are also in  pyodide web console [pyodide.org/en/stable/console.html](https://pyodide.org/en/stable/console.html). Could you please tell me whether  the kind of audio/video file download is supported in pyodide? If yes, what should I do?

>>> with youtube_dl.YoutubeDL(ydl_opts) as ydl:
        ydl.download(['https://www.youtube.com/watch?v=dQw4w9WgXcQ'])
... 
[youtube] dQw4w9WgXcQ: Downloading webpage
[0;33mWARNING:[0m Unable to download webpage: <urlopen error [Errno 23] Host is unreacha
ble>
[youtube] dQw4w9WgXcQ: Downloading API JSON
[0;31mERROR:[0m Unable to download API page: <urlopen error [Errno 23] Host is unreachab
le> (caused by URLError(OSError(23, 'Host is unreachable')))
Traceback (most recent call last):
  File "/lib/python312.zip/http/client.py", line 1081, in sendoutput
    self.send(msg)
  File "/lib/python312.zip/http/client.py", line 1025, in send
    self.connect()
  File "/lib/python312.zip/http/client.py", line 1461, in connect
    super().connect()
  File "/lib/python312.zip/http/client.py", line 991, in connect
    self.sock = self._create_connection(
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/lib/python312.zip/socket.py", line 852, in create_connection
    raise exceptions[0]
  File "/lib/python312.zip/socket.py", line 837, in create_connection
    sock.connect(sa)
OSError: [Errno 23] Host is unreachable

**Comment by hoodmane** [comment]
`urllopen` does not work #140. It's possible now with stack switching though you need to opt into [the JSPI origin trial](https://developer.chrome.com/blog/webassembly-jspi-origin-trial) to use it.
